sklearn/discriminant_analysis.py

```diff
@@ -416,7 +418,7 @@ class scatter). This solver supports both classification and
         St = _cov(X, shrinkage, covariance_estimator)  # total scatter
         Sb = St - Sw  # between scatter
 
-        evals, evecs = linalg.eigh(Sb, Sw)
+        evals, evecs = linalg.eigh(Sb, Sw, check_finite=False)
```

Comment:
For the same reason as https://github.com/scikit-learn/scikit-learn/pull/18909/files#r531112519 (usage of `_cov`) the inputs could be non finite

Reply:
Then we should be fine here as well.